lambda/index.js

```diff
@@ -1 +1,59 @@
-// TODO Implement lambda
\ No newline at end of file
+const zlib = require('zlib')
+const https = require('https')
+
+function sendLogEventsToHumio (logEvents) {
+  const data = JSON.stringify([{
+    tags: {
+      SubIdxNM: process.env.SUB_IDX_NM
+    },
+    events: logEvents.map(event => {
+      return {
+        timestamp: event.timestamp,
+        timezone: 'America/Denver',
+        attributes: JSON.parse(event.message)
+      }
+    })
+  }])
+
+  const options = {
+    hostname: (process.env.ENV === 'prd') ? 'oit-humio.byu.edu' : 'oit-humio-dev.byu.edu',
+    port: 443,
+    path: '/api/v1/ingest/humio-structured',
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'Content-Length': data.length,
+      Authorization: 'Bearer ' + process.env.HUMIO_INGEST_TOKEN
+    }
+  }
+
+  return new Promise((resolve, reject) => {
+    const req = https.request(options, function (res) {
+      if (res.statusCode >= 400) {
+        reject(new Error(`[Humio API Error] ${res.statusCode} - ${res.statusMessage}`))
+      } else {
+        resolve()
+      }
+    })
+
+    req.on('error', (error) => {
+      const errorMsg = `[HTTPS Error] ${error.name} - ${error.message}`
+      console.error(errorMsg)
+      reject(errorMsg)
+    })
+
+    req.write(data)
+    req.end()
+  })
+}
+
+exports.handler = async function (event) {
+  console.debug('Event: ' + JSON.stringify(event, null, 2))
+
+  const payload = new Buffer.from(event.awslogs.data, 'base64')
+  zlib.gunzip(payload, function (e, decodedEvent) {
+    console.debug('Decoded event: ' + decodedEvent)
+    decodedEvent = JSON.parse(decodedEvent.toString('ascii'))
+    return sendLogEventsToHumio(decodedEvent.logEvents)
+  })
+}
```

Comment:
Yeah. I just copied this code from some [AWS example code](https://github.com/awsdocs/aws-lambda-developer-guide/blob/main/sample-apps/error-processor/processor/index.js). They actually had a [check for `e` and called `context.fail(e)` if it existed](https://github.com/awsdocs/aws-lambda-developer-guide/blob/e0af1ee737a7c1ed14682d374d2ddd8e0a6cfd4a/sample-apps/error-processor/processor/index.js#L88). I'll put that back in.

Reply:
Oh. That wasn't what I was referring to, but I missed that, too.

There was a lot going on:
1) We weren't catching the `e` as you mentioned
2) If `sendLogEventsToHumio` _rejects_, our anonymous _synchronous_ callback (`function (e, decodedEvent) {}`) doesn't handle that, hence me talking about unhandled promises
3) Our top-level handler (`async function (event) {}`) doesn't care what happens after it calls `zlib.gunzip()`

The way I'd solve this would be to promisify `zlib.gunzip()`, either using [the native Node.js utility](https://nodejs.org/dist/latest-v14.x/docs/api/util.html#util_util_promisify_original) or a simple wrapper
```
function asyncGunzip (payload) {
  return new Promise((resolve, reject) => {
    zlib.gunzip(payload, (e, decodedEvent) => {
      if (e) reject(e)
      resolve(decodedEvent)
    })
  })
}
```
and then promise rejects would bubble up as expected, so they could be handled nicely in the top-level async function, which would likely `await` `asyncGunzip` and wrap it in a `try`/`catch`.